### PR TITLE
Add mtls options to aggregate drains. 

### PR DESCRIPTION
**Pull request by Benjamintf1**
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Testing performed?

- [x] Unit tests
- [x] Integration tests
- [x] Acceptance tests

## Checklist:

- [x] This PR is being made against the `main` branch, or relevant version branch
- [ ] I have made corresponding changes to the documentation
- [x] I have added testing for my changes

If you have any questions, or want to get attention for a PR or issue please reach out on the [#logging-and-metrics channel in the cloudfoundry slack](https://cloudfoundry.slack.com/archives/CUW93AF3M)

### CODE REVIEW

**Comment by Benjamintf1** [comment]
image
image
image
manifest stuff

**Comment by Benjamintf1** [comment]
image
Also this, cause have to add client_auth to certificate extended authorities for this version.

Part of me assumes we'd actual generate another cert for the real version with only client perms, but I was being lazy.

**Review by chombium — CHANGES_REQUESTED** [review]
It looks good, but I think we need few more tests.

Do you think we should also document how the aggregate drains with mTLS will work and can be configured or the syslog-binding-cache job's spec is enough?

**Comment by chombium** [comment]
Hi @Benjamintf1 a part from some small changes I think it looks good. I will do few tests this week as well and will write back.

The one thing that is not clear to me apart from specifying the certs in the binding cache is how one gets everything configured... To setup everything one should create a config file and put the certificate somewhere, so that they get picked up. Should one simply them somewhere, adjust the config and restart the syslog binding cache?

**Comment by Benjamintf1** [comment]
> Hi @Benjamintf1 a part from some small changes I think it looks good. I will do few tests this week as well and will write back.

> The one thing that is not clear to me apart from specifying the certs in the binding cache is how one gets everything configured... To setup everything one should create a config file and put the certificate somewhere, so that they get picked up. Should one simply them somewhere, adjust the config and restart the syslog binding cache? For usage with cf-d will there be an ops file provided?

Yeah, we should come up with and update the opsfile for cf-deployment for sure. There *does* already exist a aggregate drain opsfile I believe, but it'd have to be modified and or deprecated. 

To get this to work for non-log-cache usecases, I think it'd normally involve adding the certificates to credhub/a vars file and then deploying with those credentials? I'm not sure I understand.

**Comment by Benjamintf1** [comment]
As for log-cache, one of us would simply need to make a "make-log-cache-use-mtls" pr which either makes log-cache use mtls, or has a opsfile and deprecates the old regular non-mtls old format agg drain opsfile.

**Comment by Benjamintf1** [comment]
(i'll try and make some of these changes friday).

File: jobs/loggr-syslog-binding-cache/templates/aggregate_drains.yml.erb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<%
```
[the remark this answers is not in the document]
Comment (chombium) [reply jobs/loggr-syslog-binding-cache/templates/aggregate_drains.yml.erb:1]: You can add test for the spec and check if everything is ok in the bpm.yml. Something like in my [PR](https://github.com/cloudfoundry/loggregator-agent-release/pull/187/files#diff-33ba1ba1b045229c48e9617cf3131bbce369784eccdcde952a9d54bc718846fc)

[the remark this answers is not in the document]
Comment (Benjamintf1) [reply jobs/loggr-syslog-binding-cache/templates/aggregate_drains.yml.erb:1]: I tested formatting the honestly not so lazy way. I deployed the whole thing both ways.

**Comment by Benjamintf1** [comment]
So we didn't get as much work in today as I'd like. 2 open items on this pr.

1) we want to add some more app level tests to syslog_agent mtls suite for aggregate drains.
2) We think we're waiting for some final feedback from you @chombium. We tested it e2e with log-cache and it works great, and that's what you're mainly concerned about if i remember correctly. We will have to figure out how to handle deprecation of the old format of aggregate drains in cf-d when we implement mtls log-cache.

We currently arn't planning on adding spec/rubyspec tests for repo. I understand and appreciate the enthusiasm, we feel like we're adequately covered by our pipeline at the moment for now.

**Comment by Benjamintf1** [comment]
Added app tests. Interested in your feedback.

**Comment by Benjamintf1** [comment]
https://github.com/cloudfoundry/cf-deployment/pull/1068

cf-d pr

**Comment by chombium** [comment]
@Benjamintf1 Sorry for the late reply, I have to much on my plate lately :( The change looks good.

**Pull request merged**